### PR TITLE
Only send userid in Dynamic Sampling Context if sendDefaultPii is true

### PR DESCRIPTION
**Pull request by adinauer**
## :scroll: Description

See https://github.com/getsentry/develop/pull/625

## :bulb: Motivation and Context

Replaces https://github.com/getsentry/sentry-java/pull/2145 and not only skips `userId` in `baggage` but also in the envelope header `trace`.

## :pencil: Checklist

- [ ] I reviewed the submitted code
- [ ] I added tests to verify the changes
- [ ] I updated the docs if needed
- [ ] No breaking changes

### CODE REVIEW

File: sentry/src/test/java/io/sentry/SentryTracerTest.kt [hunk 1]
```diff
@@ -500,6 +501,29 @@ class SentryTracerTest {
         }
     }
 
+    @Test
+    fun `returns trace state without userId if not send pii`() {
```
Comment (philipphofmann) [inline sentry/src/test/java/io/sentry/SentryTracerTest.kt:505]: `l`: For full conditional coverage you could also add a test if the user is null and `isSendDefaultPii = true`, but maybe that's overkill. Up to you.

**Review by philipphofmann — APPROVED** [review]
LGTM

File: sentry/src/test/java/io/sentry/SentryTracerTest.kt (hunk 1 above)
Comment (adinauer) [reply sentry/src/test/java/io/sentry/SentryTracerTest.kt:505]: added `returns baggage header without userId if send pii and null user` below

**Review by philipphofmann — APPROVED** [review]
Thanks, LGTM.
